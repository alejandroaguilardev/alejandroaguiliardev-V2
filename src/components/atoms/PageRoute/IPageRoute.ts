export interface IPage {
    page:string;
    sx?:object;
}