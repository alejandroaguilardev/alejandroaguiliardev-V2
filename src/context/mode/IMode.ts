import { ReactElement } from "react";

export interface IMode {
	mode: string;
}

export interface IModeProvider {
	children:  React.ReactElement | React.ReactElement[] | React.ReactNode;
}

export interface IInit {
	theme: { mode: string };
	handleTheme: (mode: string) => void;
}
