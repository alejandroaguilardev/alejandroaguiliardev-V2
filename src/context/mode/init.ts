export const INIT = {
    mode: 'light'
}