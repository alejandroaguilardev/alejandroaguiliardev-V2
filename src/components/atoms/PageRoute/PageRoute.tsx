import Link from "next/link";
import { useRouter } from "next/router";
import { IPage } from "./IPageRoute";
import { Typography } from "@mui/material"

export const PageRoute = ({ page, sx = {} }: IPage) => {
    const router = useRouter();

    return (
        <Link href={`/${page.toLocaleLowerCase()}`}>
            <a>
                <Typography sx={{ ...sx, color: `${router.pathname === `/${page.toLocaleLowerCase()}` ? 'primary.main' : '#000'}` }}>
                    {page}
                </Typography>
            </a>
        </Link>
    )
}
