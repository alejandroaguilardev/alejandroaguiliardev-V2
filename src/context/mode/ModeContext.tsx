import { createContext } from "react";
import { IInit } from "./IMode";

export const ModeContext = createContext({} as IInit);
