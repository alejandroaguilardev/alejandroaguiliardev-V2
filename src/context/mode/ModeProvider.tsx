import { useEffect, useReducer } from "react";
import { IModeProvider } from "./IMode";
import { INIT } from "./init";
import { ModeContext } from "./ModeContext";
import { ModeReducer } from "./ModeReducer";

export const ModeProvider = ({ children }: IModeProvider) => {
	const [theme, dispatch] = useReducer(ModeReducer, INIT);

	const handleTheme = (mode: string) => {
		dispatch({
			type: "changeMode",
			payload: { mode },
		});
	};

	useEffect(() => {
		handleTheme(localStorage.getItem("mode") ?? "light");
	}, []);

	return (
		<ModeContext.Provider value={{ theme, handleTheme }}>
			{children}
		</ModeContext.Provider>
	);
};
