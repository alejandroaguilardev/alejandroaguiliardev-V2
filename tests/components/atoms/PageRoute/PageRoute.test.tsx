import { PageRoute } from "../../../../src/components/atoms/PageRoute/PageRoute";
import { render, screen } from "@testing-library/react";
import { ModeContext } from "../../../../src/context/mode/ModeContext";
import { INIT } from "../../../../src/context/mode/init";

jest.mock("next/router", () => ({
	useRouter: () => ({ pathname: "pathname" }),
}));

describe("Page Route Test", () => {
	test("toMacthSnapshot", () => {
		const page = "Page Test";
		const mockHandleTheme = jest.fn();

		const { container } = render(
			<ModeContext.Provider
				value={{ theme: INIT, handleTheme: mockHandleTheme }}
			>
				<PageRoute page={page} />
			</ModeContext.Provider>
		);

		expect(container).toMatchSnapshot();
		expect(screen.getByText(page)).toBeTruthy();
	});
});
