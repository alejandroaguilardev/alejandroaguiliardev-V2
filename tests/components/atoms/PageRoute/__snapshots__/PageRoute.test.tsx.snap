// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Route Test toMacthSnapshot 1`] = `
<div>
  <a
    href="/page%20test"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-4peds9-MuiTypography-root"
    >
      Page Test
    </p>
  </a>
</div>
`;
